utils/eeprom_metadata.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+
+# By Michael Bartlett, https://github.com/bartlettmic
+"""
+USAGE: Provide a compiled .bin file (can be generated
+with something like `avr-objcopy -I ihex $HEX_FILE -O binary
+$BIN_FILE`) and this will prepend the necessary miniboot-header
+bytes to a resulting .eeprom file which can be streamed directly
+into EEPROM to operate with miniboot. This will also nicely
+print out the generated data header data in human readable
+format.
+"""
+
+from time import time
+from binascii import crc32, hexlify, unhexlify
+from sys import argv
+from os import stat
+
+if len(argv) < 2:
+    print("No binary file provided, dying...")
+    exit(1)
+    
+binary_file = argv[1]
+
+(mode, ino, dev, nlink, uid, gid, size, atime, mtime, ctime) = stat(binary_file)
+    
+# Header protocol:
+#  0-9   = miniboot header (ABminiboot)
+#  10-19 = sketch name (provided in arguments)
+#  20-23 = timestamp of sketch creation (creation time epoch)
+#  24-27 = timestamp of eeprom write (current timestamp epoch)
+#  28-31 = CRC32 (Checksum hex value)
+#  32-33 = Length (size of binary in Hex)
+
+with open(binary_file, 'rb') as binary:
+    payload = binary.read()
+    
+    payload_size = len(payload)
+    payload_size_hex = hexlify(payload_size.to_bytes(2,'big'))
```

Comment:
please extract these positions directly to constants.

Reply:
Can you clarify what you mean by "position" and "constant"? These values depend on the binary file you are generating the header for.

Reply:
The position is the formal contract specified in the Readme. You took that constant and baked it into the code here and there. Instead, I propose extracting it in one central place (i.e.: at the beginning of the file). That way, you don't have to search for the exact position where you hard-coded it into the code.

Reply:
Sorry, I still don't quite understand. To which "contract" are you referring? I don't see a mention of the word in the README. I'm not seeing what in this block is a "position", it is just the quantity of bytes the comprises the binary file that's passed as an argument. I think a small example would make things clearer for me. If you're referring to the `2`, that is not a baked constant but just saying the value should be truncated to 2 bytes.

Reply:
ohh. I see. My bad, sorry. Yes I was referring to the 2.